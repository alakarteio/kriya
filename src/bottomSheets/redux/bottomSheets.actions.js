export const OPEN_BOTTOMSHEETS = 'OPEN_BOTTOMSHEETS'
export const open = (code, contents) => ({ type: OPEN_BOTTOMSHEETS, payload: { code, contents } })

export const CLOSE_BOTTOMSHEETS = 'CLOSE_BOTTOMSHEETS'
export const close = code => ({ type: CLOSE_BOTTOMSHEETS, payload: code })

export const TOGGLE_BOTTOMSHEETS = 'TOGGLE_BOTTOMSHEETS'
export const toggle = code => ({ type: TOGGLE_BOTTOMSHEETS, payload: code })
