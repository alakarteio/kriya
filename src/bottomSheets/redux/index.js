import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
