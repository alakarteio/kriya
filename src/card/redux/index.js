export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
