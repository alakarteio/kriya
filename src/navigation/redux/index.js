export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
