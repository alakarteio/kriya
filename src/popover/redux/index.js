export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
