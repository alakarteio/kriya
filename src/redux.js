export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
