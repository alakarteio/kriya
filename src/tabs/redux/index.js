export { default } from './tabs'
export * from './tabs.actions'
export * from './tabs.selectors'
