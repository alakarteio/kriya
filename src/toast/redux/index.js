import toast from './toast'
import * as actions from './toast.actions'
import * as selectors from './toast.selectors'

Object.assign(toast, actions, selectors)

export * from './toast.actions'
export * from './toast.selectors'
export default toast

