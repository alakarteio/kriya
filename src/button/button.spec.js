/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import mockComponent from 'misc/__mocks__/component'
import Button from './button'
import ButtonContainer from './button.container'

jest.mock('../icon', () => mockComponent('icon'))

const snap = ({ onClick = () => {}, ...rest }) => {
  const component = renderer.create(
    <Button {...rest} onClick={onClick} />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Button', () => {
  describe('graphical (JSX)', () => {
    it('should call onClick when button clicked', () => {
      const onClick = jest.fn()

      const wrapper = mount(
        <Button onClick={onClick} />,
      )
      wrapper.find('button').simulate('click')

      expect(onClick.mock.calls.length).toBe(1)
    })

    it('should call onClick when link clicked', () => {
      const onClick = jest.fn()

      const wrapper = mount(
        <Button href="link" onClick={onClick} />,
      )
      wrapper.find('a').simulate('click')

      expect(onClick.mock.calls.length).toBe(1)
    })

    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should add children', () => snap({ children: 'a child' }))
    it('should create button with link', () => snap({ link: 'http://A_LINK_TO_THE_PAST' }))
    it('should create button with link on new tab', () => snap({ href: 'http://A_LINK_TO_THE_PAST', targetBlank: true }))
    it('should create button with left icon ', () => snap({ iconLeft: 'calendar' }))
    it('should create button with right icon ', () => snap({ iconRight: 'calendar' }))
    it('should create button with right and left icon ', () => snap({ iconRight: 'calendar', iconLeft: 'calendar' }))
    it('should create a large button', () => snap({ large: true }))
    it('should create a primary button', () => snap({ primary: true }))
    it('should create a secondary button', () => snap({ secondary: true }))
    it('should create a disabled button', () => snap({ disabled: true }))
    it('should create a flat button', () => snap({ flat: true }))
    it('should create a flat primary button', () => snap({ flat: true, primary: true }))
    it('should create a flat secondary button', () => snap({ flat: true, secondary: true }))
    it('shouldnt print ink behaviour', () => snap({ ink: false }))
    it('should have a default behaviour', () => snap({}))
  })

  describe('container', () => {
    it('should dispatch a button click event', () => {
      const store = createStore(() => ({}))
      const dispatch = jest.fn()
      store.dispatch = dispatch

      const wrapper = mount(
        <Provider store={store}>
          <ButtonContainer name="button" />
        </Provider>,
      )

      wrapper.find('button').simulate('click')

      expect(dispatch.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([{ type: 'BTN_CLICKED_button' }])
    })

    it('should dispatch a link click event', () => {
      const store = createStore(() => ({}))
      const dispatch = jest.fn()
      store.dispatch = dispatch

      const wrapper = mount(
        <Provider store={store}>
          <ButtonContainer href="href" name="link" />
        </Provider>,
      )

      wrapper.find('a').simulate('click')

      expect(dispatch.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([{ type: 'BTN_CLICKED_link' }])
    })
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
