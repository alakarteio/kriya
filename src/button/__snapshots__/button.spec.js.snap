// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Button graphical (JSX) should add children 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
  a child
</button>
`;

exports[`common/Button graphical (JSX) should add custom className 1`] = `
<button
  className="button custom"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</button>
`;

exports[`common/Button graphical (JSX) should add custom style 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</button>
`;

exports[`common/Button graphical (JSX) should create a disabled button 1`] = `
<button
  className="button disabled"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
/>
`;

exports[`common/Button graphical (JSX) should create a flat button 1`] = `
<button
  className="button flat"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
/>
`;

exports[`common/Button graphical (JSX) should create a flat primary button 1`] = `
<button
  className="button flat primaryFlat"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
/>
`;

exports[`common/Button graphical (JSX) should create a flat secondary button 1`] = `
<button
  className="button flat secondaryFlat"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
/>
`;

exports[`common/Button graphical (JSX) should create a large button 1`] = `
<button
  className="button large"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</button>
`;

exports[`common/Button graphical (JSX) should create a primary button 1`] = `
<button
  className="button primary"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</button>
`;

exports[`common/Button graphical (JSX) should create a secondary button 1`] = `
<button
  className="button secondary"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</button>
`;

exports[`common/Button graphical (JSX) should create button with left icon  1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"iconLeft","primary":false,"secondary":false,"disabled":false,"children":"calendar"}}
  </div>
</button>
`;

exports[`common/Button graphical (JSX) should create button with link 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</button>
`;

exports[`common/Button graphical (JSX) should create button with link on new tab 1`] = `
<a
  className="button"
  disabled={false}
  href="http://A_LINK_TO_THE_PAST"
  onClick={[Function]}
  style={Object {}}
  target="_blank"
  title=""
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</a>
`;

exports[`common/Button graphical (JSX) should create button with right and left icon  1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"iconLeft","primary":false,"secondary":false,"disabled":false,"children":"calendar"}}
  </div>
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"iconRight","primary":false,"secondary":false,"disabled":false,"children":"calendar"}}
  </div>
</button>
`;

exports[`common/Button graphical (JSX) should create button with right icon  1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"iconRight","primary":false,"secondary":false,"disabled":false,"children":"calendar"}}
  </div>
</button>
`;

exports[`common/Button graphical (JSX) should have a default behaviour 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
>
  <canvas
    className="ink"
    height={0}
    onDragOver={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "borderRadius": "inherit",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    width={0}
  />
</button>
`;

exports[`common/Button graphical (JSX) shouldnt print ink behaviour 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type={undefined}
/>
`;
